### DISCUSSION TITLE
ColorSpaceMap does not permit Pattern value

### DISCUSSION THREAD

**Discussion by bdoubrov** [pdf-association/arlington-pdf-model#66] [General] [answered]
As of now, **ColorSpaceMap** permits only the following values of type Name: _DeviceGray_, _DeviceRGB_, _DeviceCMYK_. However, the value _Pattern_ should also be permitted as specified in ISO 32000-2, 8.6.3:

> Within a content stream, the CS or cs operator establishes the current colour space parameter in the graphics state. The operand shall always be name object, which either identifies one of the colour spaces that need no additional parameters **(DeviceGray, DeviceRGB, DeviceCMYK, or some cases of Pattern**) or shall be used as a key in the ColorSpace subdictionary of the current resource dictionary (see 7.8.3, "Resource dictionaries"). In the latter case, the value of the dictionary entry in turn shall be a colour space array or name. A colour space array shall never be inline within a content stream.

I understand that the confusion comes from Table 34 - Entries in a resource dictionary, where possible values in **ColorSpace** dictionary are defined as _"either the name of a device-dependent colour space or an array describing a colour space"_. 

Shall I raise an issue against ISO 32000-2, or there is sufficient evidence that `ColorSpaceMap.tsv` has to be updated to permit _Pattern_ value?

**Comment by petervwyatt** [comment, marked as answer]
Well spotted! Please do an errata... I want to keep Arlington as close as possible to "official wording" rather than "accepted practice"

**Reply by petervwyatt** [reply]
Please explicitly mention and link to this discussion in the errata so I can remember to maintain Arlington when it gets resolved.
